nixos/modules/services/networking/ntp/ntpd-rs.nix

```
@@ -0,0 +1,60 @@
+{ lib, config, pkgs, ... }:
+
+let
+  cfg = config.services.ntpd-rs;
+  configFile = pkgs.writers.writeTOML "ntpd-rs.toml" cfg.config;
+in
+{
+  options.services.ntpd-rs = {
+    enable = lib.mkEnableOption "Network Time Service (ntpd-rs)";
+    metrics.enable = lib.mkEnableOption "ntpd-rs Prometheus Metrics Exporter";
+    useNetworkingTimeServers = lib.mkOption {
+      type = lib.types.bool;
+      default = true;
+      description = lib.mdDoc ''
+        Use source time servers from NixOS option networking.timeServers in config.
+      '';
+    };
+    config = lib.mkOption {
+      type = lib.types.attrs;
+      description = lib.mdDoc "Config rendered to TOML.";
+    };
+  };
+
+  config = lib.mkMerge [
+    (lib.mkIf cfg.enable {
+      environment.systemPackages = [ pkgs.ntpd-rs ];
+      systemd.packages = [ pkgs.ntpd-rs ];
+
+      services.timesyncd.enable = lib.mkForce false;
+      systemd.services.systemd-timedated.environment = { SYSTEMD_TIMEDATED_NTP_SERVICES = "ntpd-rs.service"; };
+
+
+      systemd.services.ntpd-rs = {
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          User = "";
+          Group = "";
+          DynamicUser = true;
+          ExecStart = [ "" "${pkgs.ntpd-rs}/bin/ntp-daemon --config=${configFile}" ];
+        };
+      };
+
+      systemd.services.ntp-rs-metrics = lib.mkIf cfg.metrics.enable {
+        wantedBy = [ "multi-user.target" ];
+        serviceConfig = {
+          User = "";
+          Group = "";
+          DynamicUser = true;
+          ExecStart = [ "" "${pkgs.ntpd-rs}/bin/ntp-metrics-exporter --config=${configFile}" ];
+        };
+      };
+    })
+    (lib.mkIf (cfg.enable && cfg.useNetworkingTimeServers) {
+      services.ntpd-rs.config.source = lib.flip builtins.map config.networking.timeServers (ts: {
+        mode = "server";
+        address = ts;
+      });
+    })
+  ];
+}
```

Would you mind maintaining the module?

```suggestion

  meta.maintainers = with lib.maintainers; [ fpletz ];
}
```